c7n/query.py

```diff
@@ -625,9 +635,9 @@ def get_arns(self, resources):
         id_key = m.id
 
         for r in resources:
-            _id = r[id_key]
+            _id = jmespath_search(id_key, r)
```

Comment:
preference would be to to implement the get_arns method in the subclass versus putting it on the base, as jmespath execution is actually relatively expensive on cost, versus a key lookup for a high cardinality resources. ie only pay the cost when its actually needed by putting the logic specific to the resource types that need it.